.github/workflows/phpunit-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: PHPUnit Tests
+
+# Since Unit Tests are required to pass for each PR,
+# we cannot disable them for documentation-only changes.
+on:
+    pull_request:
+    push:
+        branches:
+            - trunk
+            - 'release/**'
+            - 'wp/**'
+
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+    # The concurrency group contains the workflow name and the branch name for pull requests
+    # or the commit hash for any other events.
+    group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+    cancel-in-progress: true
+
+jobs:
+    # Runs the PHPUnit tests for Gutenberg.
+    #
+    # Performs the following steps:
+    # - Sets environment variables.
+    # - Checks out the repository.
+    # - Sets up Node.js.
+    # - Sets up PHP.
+    # - Installs Composer dependencies.
+    # - Installs npm dependencies
+    # - Logs general debug information about the runner.
+    # - Logs Docker debug information (about the Docker installation within the runner).
+    # - Starts the wp-env Docker containers.
+    # - Logs the running Docker containers.
+    # - Logs debug information about what's installed within the wp-env Docker containers.
+    # - Install WordPress and Gutenberg within the Docker container.
+    # - Run the PHPUnit tests.
+    test-php:
+        name: ${{ matrix.php }}${{ matrix.multisite && ' multisite' || '' }} on ${{ matrix.os }}
+        runs-on: ${{ matrix.os }}
```

Comment:
Why have the `os` as a matrix value and in the job name if only one OS is being used ?

Reply:
This code has been copied from the corresponding Core.
I agree. It makes no sense to have it as a matrix value.
Fixed in 8210cb28e67787e8172ea686dc2ef1825eb5e1ae